File: Code_CS/C18_AppLogic/App_Code/GlobalMembers.cs
using System;
using System.IO;

public class GlobalMembers
{
    public static int successRate = 50;

    public GlobalMembers()
    {
    }

    protected void Application_Start(Object sender, EventArgs e)
    {
        WriteFile("Application Starting");
    }

    protected void Application_End(Object sender, EventArgs e)
    {
        WriteFile("Application Ending");
    }

    public void WriteFile(string strText)
    {
        using (StreamWriter writer =
           new StreamWriter(@"C:\Users\Public\test.txt", true))
        {
            string str;
            str = DateTime.Now.ToString() + " " + strText;
            writer.WriteLine(str);
            writer.Close();
        }
    }

    public static void StaticWriteFile(string strText)
    {
        using (StreamWriter writer =
           new StreamWriter(@"C:\Users\Public\test.txt", true))
        {
            string str;
            str = DateTime.Now.ToString() + " " + strText;
            writer.WriteLine(str);
            writer.Close();
        }
    }
}

File: Code_VB/C18_AppLogic/App_Code/GlobalMembers.vb
Imports System
Imports System.IO

Public Class GlobalMembers
    Public Shared successRate As Integer = 50

    Public Sub New()
    End Sub

    Protected Sub Application_Start(ByVal sender As Object, ByVal e As EventArgs)
        WriteFile("Application Starting")
    End Sub

    Protected Sub Application_End(ByVal sender As Object, ByVal e As EventArgs)
        WriteFile("Application Ending")
    End Sub

    Public Sub WriteFile(ByVal strText As String)
        Using writer As New StreamWriter(HttpContext.Current.Server.MapPath("test.txt"), True)
            Dim str As String
            str = (DateTime.Now.ToString() & " ") + strText
            writer.WriteLine(str)
            writer.Close()
        End Using
    End Sub

    Public Shared Sub StaticWriteFile(ByVal strText As String)
        Using writer As New StreamWriter(HttpContext.Current.Server.MapPath("test.txt"), True)
            Dim str As String
            str = (DateTime.Now.ToString() & " ") + strText
            writer.WriteLine(str)
            writer.Close()
        End Using
    End Sub
End Class